### PR TITLE
[charts/cosi] add COSI driver helm chart

### PR DESCRIPTION
**Pull request by mdutka-dell**
#### Is this a new chart?

Yes

#### What this PR does / why we need it:

Adding COSI driver to the helm-charts repo.

#### Which issue(s) is this PR associated with:

KRV-16397

#### Special notes for your reviewer:

Adding COSI driver / Kube Linter GitHub Action.

#### Checklist:

[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]

- [ ] Chart Version bumped
- [ ] Variables are documented in the chart README.md
- [x] Title of the PR starts with the chart name (e.g. `[charts_dir/mychartname]`) if applicable

### CODE REVIEW

**Review by bogdanNovikovDell — DISMISSED** [review]

File: .github/workflows/cosi-helm.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    branches: [main]
+  pull_request:
+    branches: ["**"]
+
+env:
+  GOPRIVATE: github.com/dell/*
+  TOKEN: ${{ secrets.GH_DELL_ACCESS }}
+
+jobs:
+  kube-linter:
```
Comment (tdawe) [inline .github/workflows/cosi-helm.yaml:14]: There is an existing linter used for all charts located here: https://github.com/dell/helm-charts/blob/main/.github/workflows/helm-validations.yml#L29C6-L29C6

This file may no longer be required or that existing helm-validations.yml file could be updated if this new kube-linter-action is more up to date than what we are using in our existing workflow.

Comment (mdutka-dell) [reply .github/workflows/cosi-helm.yaml:14]: As you can see here: https://github.com/dell/helm-charts/actions/runs/5999804015/job/16270649466?pr=267, after adding scanning to the whole repo many issues in other charts appeared. 

I think that in this PR the cosi chart should be scanned, but other teams should correct their charts. The additional tasks have to be planned.

Comment (shanduur-dell) [reply .github/workflows/cosi-helm.yaml:14]: @tdawe To add my 5 cents, I think there should be 2 parallel configurations for KubeLinter, as the rules applicable to things like COSI Driver or CS Modules, might not be applicable to CSI Drivers, as CSI Drivers are interacting with K8s Node's filesystem.

**Review by shanduur-dell — APPROVED** [review]

File: charts/cosi/templates/_helpers.tpl [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+# Default value: "debug"
+*/}}
+{{- define "cosi.logLevel" }}
+  {{- $logLevelValues := list "trace" "debug" "info" "warn" "error" "fatal" "panic" }}
+  {{- if (has .Values.provisioner.logLevel $logLevelValues) }}
+    {{- .Values.provisioner.logLevel }}
+  {{- else }}
+    {{- "debug" }}
+  {{- end }}
+{{- end }}
```
Comment (shanduur-dell) [inline charts/cosi/templates/_helpers.tpl:45]: Fix this to reflect numeric values.

**Review by shanduur-dell — CHANGES_REQUESTED** [review]

**Review by shanduur-dell — APPROVED** [review]

**Review by mtrajfacki-dell — APPROVED** [review]

**Review by shanmydell — APPROVED** [review]

**Pull request merged**